cmd/setec/setec.go

```diff
@@ -140,3 +212,154 @@ func runServer(ctx context.Context, args []string) error {
 
 	return nil
 }
+
+func clientFromArgs(args []string) (client *setec.Client, remainingArgs []string, err error) {
+	if len(args) < 1 {
+		return nil, nil, flag.ErrHelp
+	}
+	server := args[0]
+	// TODO(corp/13375): make a better UX here where single-label
+	// hostnames get expanded into a full HTTPS URL.
+	return &setec.Client{
+		Server: server,
+	}, args[1:], nil
+}
+
+func runList(ctx context.Context, args []string) error {
+	c, args, err := clientFromArgs(args)
+	if err != nil {
+		return err
+	}
+	if len(args) > 0 {
+		return flag.ErrHelp
+	}
+
+	secrets, err := c.List(ctx)
+	if err != nil {
+		return fmt.Errorf("Failed to list secrets: %v", err)
+	}
+
+	tw := tabwriter.NewWriter(os.Stdout, 0, 8, 0, '\t', 0)
+	io.WriteString(tw, "NAME\tACTIVE\tVERSIONS\n")
+	for _, s := range secrets {
+		vers := make([]string, 0, len(s.Versions))
+		for _, v := range s.Versions {
+			vers = append(vers, v.String())
+		}
+		fmt.Fprintf(tw, "%s\t%s\t%s\n", s.Name, s.ActiveVersion, strings.Join(vers, ","))
+	}
+	if err := tw.Flush(); err != nil {
+		return err
+	}
+	return nil
+}
+
+var getArgs struct {
+	Version uint64
+}
+
+func runGet(ctx context.Context, args []string) error {
+	c, args, err := clientFromArgs(args)
+	if err != nil {
+		return err
+	}
+	if len(args) != 1 {
+		return flag.ErrHelp
+	}
+
+	name := args[0]
+
+	var val *api.SecretValue
+	if getArgs.Version == 0 {
+		val, err = c.Get(ctx, name)
+	} else {
+		val, err = c.GetVersion(ctx, name, api.SecretVersion(getArgs.Version))
+	}
+	if err != nil {
+		return fmt.Errorf("Failed to get secret: %v", err)
```

Comment:
(or honestly I'd be fine with just fataling directly from in here, since these functions are "morally main". Or maybe [ctrl](https://godoc.org/github.com/creachadair/ctrl) 😄)

Reply:
For now, fixed by lowercasing all the errors, and we can figure something out if that's too hateful to stick to.

Can I also say I love that a wild "dynamic extent" appears in the docs of ctrl :) An elegant weapon of a more civilized age.